### DISCUSSION TITLE
Questions about Hyper/Terminal configuration + Git + technical documentation creation

### DISCUSSION THREAD

**Discussion by TCG85** [Pixel-Prowess/GitHub360-Summer24#2] [General]
Hi @planetoftheweb 

I hope this post finds you well!

I'm having the following four questions with regards to our course:

1. When using Hyper as a terminal, you had the branches displayed + when using "ls" command, it showed more information compared to the default configuration (by default "ls" just shows a list of the files, to display more information you need to use "ls -la" for example). How do I enable Hyper to have the same settings/configuration as you had?

2. What is the best/easiest way to hide hostname and username in Terminal? Is there any better way than what's described here https://www.youtube.com/watch?v=ydQTeIhH6fU ?

3. In case of working offline, once I connected a local repo to a remote location to push my code for the first time (using the GitHub remote commands git remote add origin  + git push -u origin main), will I need to do this step again the next time I have worked on my code and would like to push it?

5. You described how to create technical documentation with the help of ChatGPT. Let's assume a more complex case where you have a lot of code across various files that have dependencies between each other etc. 
a) Is there any way to give a whole repo as an input to ChatGPT? Or is there any other tool that is capable of that?
b) in case the project is private, how can you ensure that the code is not used to train ChatGPT?

Many thanks,
T

**Comment by planetoftheweb** [comment]
> When using Hyper as a terminal, you had the branches displayed + when using "ls" command, it showed more information compared to the default configuration (by default "ls" just shows a list of the files, to display more information you need to use "ls -la" for example). How do I enable Hyper to have the same settings/configuration as you had?

That's correct. You don't mean the branches though, branches are displayed when you do the `git branch` command. However the files are listed with the ls command. To have a certain look to your default ls command, you need to edit a file when your shell loads. In my case, I'm using `zshell` with hyper, so the file is named `.zshrc`. If you're using bash, then the file is named `.bashrc` or something like that.

To edit the file, I create an alias in that file.

`alias ls='ls -1sAhFG'`

Here's what all those letters [mean](https://www.man7.org/linux/man-pages/man1/ls.1.html).

> What is the best/easiest way to hide hostname and username in Terminal? Is there any better way than what's described here https://www.youtube.com/watch?v=ydQTeIhH6fU ?

That works, I think it might depend on the shell you're using. I think mine is `PROMPT="%# "` for zshell.

Next week, I hope to show you how to control that inside the codespaces so they load up more configured to what you need.

> In case of working offline, once I connected a local repo to a remote location to push my code for the first time (using the GitHub remote commands git remote add origin + git push -u origin main), will I need to do this step again the next time I have worked on my code and would like to push it?

No. You should only have to do that the first time you set things up. That's what the -u option does, it makes the connection permanent for that instance of the remote.

> You described how to create technical documentation with the help of ChatGPT. Let's assume a more complex case where you have a lot of code across various files that have dependencies between each other etc.
> a) Is there any way to give a whole repo as an input to ChatGPT? Or is there any other tool that is capable of that?
> b) in case the project is private, how can you ensure that the code is not used to train ChatGPT?

a) A couple of things. If you start the command with the words @workspace, then GitHub can see the contents of your repo. that feature is pretty new, so it's not perfect, but it does show you where things are going.

If you were using GitHub Enterprise, it also has the ability to access your entire codebase, not just the current repo.

There is also a new feature I haven't had enough chance to play with called Workspaces. Think of it as a way to ask GitHub to look at your whole codebase and manage your work by creating issues and planning the work for you. Even being able to suggest changes that should be made.

b) Well, I can't ensure that personally, so I'll point you to the [FAQs](https://github.com/features/copilot) on GitHub copilot's pages. However based on my understanding, it looks like GitHub does use your interactions to train the model on the general plan. but not in the Business or enterprise plans. The details specify that the code that it trains on is 'publicly available' which would lead me to believe that if your code is in a private repo with a copilot business plan that it would not train on it.
